layer/decorators/settings.py

```diff
@@ -69,31 +81,24 @@ def set_pip_requirements_file(self, file: Optional[str]) -> None:
     def set_pip_packages(self, packages: Optional[List[str]]) -> None:
         self._pip_packages = packages
 
-    def set_paths(self, paths: Optional[List[str]]) -> None:
-        self._paths = paths
-
-    def set_asset_type(self, asset_type: AssetType) -> None:
-        self._asset_type = asset_type
-
-    def set_asset_name(self, name: str) -> None:
-        self._name = name
+    def set_resource_paths(self, paths: Optional[List[ResourcePath]]) -> None:
+        self._resource_paths = paths
 
     def set_dependencies(self, dependencies: List[AssetPath]) -> None:
         self._dependencies = dependencies
 
-    def append_assertions(self, assertions: List[Any]) -> None:
-        existing_assertions = self.get_assertions()
-        existing_assertions.append(assertions)
-        self._assertions = existing_assertions
+    def append_assertion(self, assertion: Assertion) -> None:
+        self._assertions = self._assertions or []
+        self._assertions.append(assertion)
 
     def validate(self) -> None:
-        if self.get_asset_type() is None:
+        if self._asset_type is None:
```

Comment:
It's weird -- we have a validation in both `get_asset_type()` and here. Seems fishy.